### DISCUSSION TITLE
Generation/build date in the footer

### DISCUSSION THREAD

**Discussion by PlasmaHH** [squidfunk/mkdocs-material#7300] [Q&A] [answered]
Hi, after trying out the git-revision-date-localized plugin we found it to be a bit too slow for our needs ( well more like git at that point being too slow, either way the build takes 30 minutes instead of 3) we agreed that the most intresting information anyways is the build timestamp (to figure out if one should be able to see the latest push already).

The information needs to be in the footer.

So I tried finding something and googling for it seems to be rather hard... I know I could write a plugin myself that does nothing more than to provide a python datetime object for jinja this seems to be a bit excessive, especially since its nontrivial to provide it with my documentation.

So the question is, is there anywhere a date object or so that one could use? Or would it possible to just add one on the the jinja environment as a global? probably just a oneliner somewhere appropriate: `jinja_env["date"] = datetime.now()` or so....

**Comment by squidfunk** [comment, marked as answer]
I think [`build_date_utc`](https://www.mkdocs.org/dev-guide/themes/#build_date_utc) and a [partial override](https://squidfunk.github.io/mkdocs-material/customization/#overriding-partials) (footer?) should tick all boxes.

**Reply by PlasmaHH** [reply]
Ah of course, I must have somehow mixed that up with `page.update_date`, thanks a lot!

**Reply by PlasmaHH** [reply]
Ok, not sure if that is worth a topic elsewhere (in a jinja place maybe) but I guess others wonder too, does anyone happen to know a way to convert the timezone? astimezone() takes a tzinfo and all hits I could find for jinja implement their own filter for that...

**Reply by squidfunk** [reply]
You could register an additional filter in `on_env` by using the `env` variable, like we do it in the blog plugin (see below). Other than that, MkDocs doesn't directly allow to do that via configuration, you would either need to implement a hook or plugin.

https://github.com/squidfunk/mkdocs-material/blob/12a8e82837fe400dd1f123e41d75b32987a11744/src/plugins/blog/plugin.py#L302-L324
